IPython/core/completer.py

```diff
@@ -117,9 +117,16 @@ def has_open_quotes(s):
 
 def protect_filename(s):
     """Escape a string to protect certain characters."""
+    if set(s) & set(PROTECTABLES):
+        if sys.platform == "win32":
+            return '"' + s + '"'
+        else:
+            return "".join(("\\" + c if c in PROTECTABLES else c) for c in s)
+            return s.translate(
```

Comment:
Double return here. I think the first way is probably a bit clearer, because `str.translate` is not very commonly used.